packages/story-editor/src/components/library/panes/pageTemplates/karma/savedPageTemplates.karma.js

```diff
@@ -118,6 +118,7 @@ describe('CUJ: Page Templates: Custom Saved Templates', () => {
       await fixture.events.hover(
         fixture.editor.library.pageTemplatesPane.pageTemplates[0]
       );
+      await fixture.events.sleep(100);
```

Comment:
Just curious, given the recent efforts to clean up sleeps, is this absolutely necessary to sleep? What changed in the flow that's making it newly needed?

Reply:
If we're waiting for the delete button to be available, perhaps we could add `waitFor` for `fixture.editor.library.pageTemplatesPane.deleteBtnByIndex(0)` where we throw an error explicitly if the button node is not available (as it's now done elsewhere instead of sleeping).

Reply:
Hmm, it looks like the delete button is available, but not actionable, even if I replace `sleep` with 
```
await waitFor(() => {
    expect(fixture.editor.library.pageTemplatesPane.deleteBtnByIndex(0).node).toBeTruthy();
});
```
it will fail.

Reply:
What about explicitly throwing an error if the node is not ready? If there's no error in the `waitFor`, it doesn't actually wait. Something like:

```
await waitFor(() => {
    const button = fixture.editor.library.pageTemplatesPane.deleteBtnByIndex(0).node;
    if (!button) {
      throw new Error('node not ready');
    }
    expect(fixture.editor.library.pageTemplatesPane.deleteBtnByIndex(0).node).toBeTruthy();
});
```

Reply:
Ah, okay, just now realized you mentioned the button itself is available but not actionable. Hm, what's the error it fails with?

Reply:
It errors at the next step (finding the delete dialog).

Reply:
What if you bring clicking and checking for the dialogue into the same `waitFor` as well, for example:
```
...
await waitFor(() => {
    const button = fixture.editor.library.pageTemplatesPane.deleteBtnByIndex(0).node;
    if (!button) {
      throw new Error('node not ready');
    }
    await fixture.events.click(
        fixture.editor.library.pageTemplatesPane.deleteBtnByIndex(0)
      );
    const dialog = await fixture.screen.getByRole('dialog', {
        name: 'Delete Page Template',
      });
    if (!dialog) {
      throw new Error('node not ready');
    }
}, { timeout: 9000 });
...
```

Reply:
> What if you bring clicking and checking for the dialogue into the same `waitFor` as well, for example:

For me, it just throws `node not ready` and that's it.